src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+data ExistsRE (f :: # ! -> *)
```

Comment:
@garyb I don't think that hiding the constructor export would help in PureScript. In ML, the reason `SIG1.welp` isn't necessarily the empty type is that the identity of the type is literally not exposed, and so you really have no idea how it is implemented (i.e. is it a datatype? or a type synonym?). 

Haskell may have the same problem, fwiw, as PureScript here, but I'm not certain. I think the proper thing to do in the language is to introduce abstraction.

In the present case, if your `newtype` idea works, then let's do that! :smile: If it's necessary on the other hand to *implement* this using an empty type + coercions, it would better express the intent if we made `newtype Welp = Welp Empty`, and then hid the constructor; I'm not certain if this ends up meaning anything different in PureScript, but I think it would at least be more clear that it is not intended to support pattern matching.